lib/models/project.js

```diff
@@ -553,18 +558,38 @@ Project.closestSync = function(pathName, _ui) {
   @param  {UI} ui The UI instance to provide to the created Project.
   @return {Project}         Project instance
  */
-Project.projectOrNullProject = function(_ui) {
+Project.projectOrNullProject = function(_ui, _cli) {
   try {
-    return Project.closestSync(process.cwd(), _ui);
+    return Project.closestSync(process.cwd(), _ui, _cli);
   } catch(reason) {
     if (reason instanceof Project.NotFoundError) {
-      return Project.NULL_PROJECT;
+      return Project.createNullProject(_ui, _cli);
     } else {
       throw reason;
     }
   }
 };
 
+/**
+  Returns the project root based on the first package.json that is found
+
+  @return {String} The project root directory
+ */
+Project.getProjectRoot = function () {
+  try {
+    var directory = findup.sync(process.cwd(), 'package.json');
+    var pkg = require(path.join(directory, 'package.json'));
+
+    if (pkg && pkg.name === 'ember-cli') {
+      return process.cwd();
+    }
+
+    return directory;
+  } catch(reason) {
+    return process.cwd();
```

Comment:
I would prefer we explicitly only swallow the error if it is an error we match against. Basically I want to be sure we are not swallowing unexpected errors.

This is likely also a good plays to add some `debug` logging.